querydsl-codegen/src/main/java/com/querydsl/codegen/EntitySerializer.java

```diff
@@ -315,7 +326,7 @@ protected void introClassHeader(CodeWriter writer, EntityType model) throws IOEx
     }
 
     protected void introDefaultInstance(CodeWriter writer, EntityType model, String defaultName) throws IOException {
-        String simpleName = !defaultName.isEmpty() ? defaultName : model.getUncapSimpleName();
+        String simpleName = !defaultName.isEmpty() ? defaultName : variableNameFunction.apply(model);
```

Comment:
You need to preserve the escaping functionality of getUncapSimpleName(), that's the reason for the QPublic and QPrivate issues (public and private can't be used as variable names)